### ISSUE TITLE
publicPath issue when code-splitting in production

### ISSUE THREAD

**Issue by kmcaloon** [10up/10up-toolkit#213]
### Describe the bug

Looks like other projects have had an issue with this in Webpack v5, mostly but not exclusively related to mini-css-extract-plugin. I am lazy loading a couple of React components perfectly fine in development, but after building and pushing to staging my app doesn't load and I get the following error: `Automatic publicPath is not supported in this browser`. The issue seems to be that because `publicPath` is not being explicitly set, the app doesn't know where to fetch the chunks. It is happening in all browsers.

There are a couple of workarounds people are using for this. Many simply set `config.output.publicPath: ''`, but I'm pretty sure this only works if your js directory lives at the root of the app. I was able to quickly work around this by setting `publicPath` to my dist directory which is being determined by custom env arguments I am already using. 

Not sure what the best solution might be to cover all sorts of use cases people might run into. Here are some related issues/solutions:

http://newspatrak.com/javascript/webpack5-automatic-publicpath-is-not-supported-in-this-browser/
https://github.com/cypress-io/cypress/issues/18435
https://discourse.roots.io/t/automatic-publicpath-is-not-supported-in-this-browser/22619/5

### Steps to Reproduce

A simple react app would need to be created that lazy loads a component.  You could include the script anywhere within your templates. After running build and pushing to a staging environment you should see the error. I wasn't able to reproduce the issue locally, only after actually pushing to staging. I didn't have time to dig in further as to why webpack might not be having this issue locally.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Antonio-Laguna** [comment]
@kmcaloon thanks for flagging this, we'll look into it!

**Comment by fabiankaegy** [comment]
I had run into a similar issue before where I had a site deploy to multiple different subpath domains. In that case it also wasn't possible to use the auto detected publicPath. 

This was not in the react context as it appears to be here. We encountered the issue in a WordPress Subsite Multisite. 

The solution we've ended up with is using `wp_localize_script` to localize the base url of the site to the frontend. In the main frontend javascript file we then set the publicPath like so:

```js
__webpack_public_path__ = window.localizedNamespace.distDirectoryUrl;
```

Hope this helps.
